netbox/extras/api/mixins.py

```diff
@@ -61,14 +63,23 @@ def render_configtemplate(self, request, configtemplate, context):
 
 
 class RenderConfigMixin(ConfigTemplateRenderMixin):
+    """
+    Override initial() to save a copy of the queryset for "un-restricting" the queryset when rendering.
+    """
+    def initial(self, request, *args, **kwargs):
+        self.original_queryset = self.queryset
+        super().initial(request, *args, **kwargs)
```

Comment:
We could alternatively modify:

```python
class BaseViewSet(GenericViewSet):
    """
    Base class for all API ViewSets. This is responsible for the enforcement of object-based permissions.
    """
    def initial(self, request, *args, **kwargs):
        super().initial(request, *args, **kwargs)

        # Restrict the view's QuerySet to allow only the permitted objects
        if request.user.is_authenticated:
            if action := HTTP_ACTIONS[request.method]:
                self.queryset = self.queryset.restrict(request.user, action)
```
to account for permissions using:

self.get_permissions() and using the resulting perms_map to determine what permission should be applied.

This is a down the road thing IMO though.

Reply:
We'll definitely need to identify a more maintainable solution. This will probably entail digging into DRF permissions logic a bit.

Reply:
As mentioned above, I think we can figure out something by modifying the `BaseViewSet` and then adjusting the queryset based on that.

I will take a second crack at this on the weekend.